vpr/src/util/vpr_utils.cpp

```diff
@@ -2191,3 +2192,19 @@ bool is_node_on_tile(t_physical_tile_type_ptr physical_tile,
         }
     }
 }
+
+void apply_route_constraints(VprConstraints& vpr_constraint) {
```

Comment:
Avoid to use global variables inside function. Instead, suggest to add them as an input argument. 
For instance (in your case): 

```
void apply_route_constraints(VprConstraints& vpr_constraint, ClusteringContext& cluster_ctx) {

}
```